components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/AuthenticatorEndpointConfiguration.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    /**
+     * AuthenticatorEndpointConfiguration builder.
+     */
+    public static class AuthenticatorEndpointConfigurationBuilder {
+
+        private String uri;
+        private String authenticationType;
+        private HashMap<String, String> authenticationProperties;
+
+        public AuthenticatorEndpointConfigurationBuilder() {
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder uri(String uri) {
+
+            this.uri = uri;
+            return this;
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder authenticationProperties(
+                HashMap<String, String> authentication) {
+
+            this.authenticationProperties = authentication;
+            return this;
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder authenticationType(String authenticationType) {
+
+            this.authenticationType = authenticationType;
+            return this;
+        }
+        
+        public AuthenticatorEndpointConfiguration build() {
+
+            try {
+                EndpointConfig.EndpointConfigBuilder actionEndpointConfigBuilder =
+                        new EndpointConfig.EndpointConfigBuilder();
+                actionEndpointConfigBuilder.uri(this.uri);
+                actionEndpointConfigBuilder.authentication(buildAuthentication(
+                        this.authenticationType, this.authenticationProperties));
+                actionEndpointConfigBuilder.build();
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Endpoint configuration of the authenticator is not in expected " +
+                        "format.", e);
+            }
+            return new AuthenticatorEndpointConfiguration(this);
+        }
+        
+        private Authentication buildAuthentication(String authenticationType,
+                        HashMap<String, String> properties) throws NoSuchElementException {
+
+            switch (Authentication.Type.valueOf(authenticationType)) {
+                case BASIC:
+                    return new Authentication.BasicAuthBuilder(
+                            getProperty(properties, Property.USERNAME.getName()),
+                            getProperty(properties, Property.PASSWORD.getName())).build();
+                case BEARER:
+                    return new Authentication.BearerAuthBuilder(
+                            getProperty(properties, Property.ACCESS_TOKEN.getName())).build();
+                case API_KEY:
+                    return new Authentication.APIKeyAuthBuilder(
+                            getProperty(properties, Property.HEADER.getName()),
+                            getProperty(properties, Property.VALUE.getName())).build();
+                case NONE:
+                    return new Authentication.NoneAuthBuilder().build();
+                default:
+                    throw new IllegalArgumentException();
+            }    
+        }
```

Comment:
Is this necessary ? Doesn't the action management endpoint model handles this ?

Reply:
No, in the action management model provide only builder per authentication Type.

Reply:
Add this logic to the authentication model.